packages/hardhat/contracts/strategy/CrabStrategyV2.sol

```diff
@@ -722,188 +572,156 @@ contract CrabStrategyV2 is StrategyBase, StrategyFlashSwap, ReentrancyGuard, Own
-            _exactOutFlashSwap(
-                wPowerPerp,
-                weth,
-                IUniswapV3Pool(ethWSqueethPool).fee(),
-                ethProceeds,
-                wSqueethToAuction,
-                uint8(FLASH_SOURCE.FLASH_HEDGE_SELL),
-                abi.encodePacked(wSqueethToAuction, ethProceeds, _minWSqueeth, _minEth)
-            );
-        } else {
-            _exactOutFlashSwap(
-                weth,
-                wPowerPerp,
-                IUniswapV3Pool(ethWSqueethPool).fee(),
-                wSqueethToAuction,
-                ethProceeds,
-                uint8(FLASH_SOURCE.FLASH_HEDGE_BUY),
-                abi.encodePacked(wSqueethToAuction, ethProceeds, _minWSqueeth, _minEth)
-            );
-        }
+    function nonces(address owner) external view returns (uint256) {
+        return _nonces[owner].current();
+    }
 
-        emit HedgeOnUniswap(msg.sender, isSellingAuction, auctionWSqueethEthPrice, wSqueethToAuction, ethProceeds);
+    /**
+     * @dev view function to get the domain seperator used in signing
+     */
+    // solhint-disable-next-line func-name-mixedcase
+    function DOMAIN_SEPARATOR() external view returns (bytes32) {
+        return _domainSeparatorV4();
     }
 
     /**
-     * @notice execute sell auction based on the parameters calculated
-     * @dev if _isHedgingOnUniswap, wSqueeth minted is kept to repay flashswap, otherwise sent to seller
-     * @param _buyer buyer address
-     * @param _buyerAmount buyer ETH amount sent
-     * @param _wSqueethToSell wSqueeth amount to sell
-     * @param _ethToBuy ETH amount to buy
-     * @param _isHedgingOnUniswap true if arbing with uniswap price
+     * @dev check the signer and swap tokens in the order
+     * @param managerSellAmount quantity the manager wants to sell
+     * @param managerBuyPrice the price at which the manager is buying
+     * @param _order a signed order to swap tokens
      */
-    function _executeSellAuction(
-        address _buyer,
-        uint256 _buyerAmount,
-        uint256 _wSqueethToSell,
-        uint256 _ethToBuy,
-        bool _isHedgingOnUniswap
+    function _execOrder(
+        uint256 managerSellAmount,
+        uint256 managerBuyPrice,
+        uint256 sellerPrice,
+        Order memory _order
     ) internal {
-        if (_isHedgingOnUniswap) {
-            _mintWPowerPerp(_buyer, _wSqueethToSell, _ethToBuy, true);
-        } else {
-            _mintWPowerPerp(_buyer, _wSqueethToSell, _ethToBuy, false);
+        require(managerBuyPrice >= sellerPrice, "Manager Buy Price should be atleast Seller Price");
+        bytes32 structHash = keccak256(
+            abi.encode(
+                _CRAB_BALANCE_TYPEHASH,
+                _order.bidId,
+                _order.trader,
+                _order.traderToken,
+                _order.traderAmount,
+                _order.managerToken,
+                _order.managerAmount,
+                _useNonce(_order.trader)
```

Comment:
We don't do any nonce validation or even use it right? Shouldn't we check that the nonce provided is <= to current nonce?

Reply:
we do validate it, useNonce is the current nonce of the trader. So a nonce can be used only once, and when it is used it is incremented.
If that is not the case, the signature will not match. i.e the signer will not be equal to the trader